src/anemoi/models/models/encoder_processor_decoder.py

```diff
@@ -239,26 +245,29 @@ def forward(self, x: Tensor, model_comm_group: Optional[ProcessGroup] = None) ->
         # add skip connection (hidden -> hidden)
         x_latent_proc = x_latent_proc + x_latent
 
-        # Run decoder
-        x_out = self._run_mapper(
-            self.decoder,
-            (x_latent_proc, x_data_latent),
-            batch_size=batch_size,
-            shard_shapes=(shard_shapes_hidden, shard_shapes_data),
-            model_comm_group=model_comm_group,
-        )
+        # Run decoders
+        x_out = {}
+        for data, decoder in self.decoders.items():
+            x_out[data] = self._run_mapper(
+                decoder,
+                (x_latent_proc, x_data_latent[data]),
+                batch_size=batch_size,
+                shard_shapes=(shard_shapes_hidden, shard_shapes_data[data]),
+                model_comm_group=model_comm_group,
+            )
 
-        x_out = (
-            einops.rearrange(
-                x_out,
-                "(batch ensemble grid) vars -> batch ensemble grid vars",
-                batch=batch_size,
-                ensemble=ensemble_size,
+            x_out[data] = (
+                einops.rearrange(
+                    x_out[data],
+                    "(batch ensemble grid) vars -> batch ensemble grid vars",
+                    batch=batch_size,
+                    ensemble=ensemble_size,
+                )
+                .to(dtype=x.dtype)
+                .clone()
             )
-            .to(dtype=x.dtype)
-            .clone()
-        )
 
-        # residual connection (just for the prognostic variables)
-        x_out[..., self._internal_output_idx] += x[:, -1, :, :, self._internal_input_idx]
-        return x_out
+            # residual connection (just for the prognostic variables)
+            if data in self._graph_input_meshes:
+                x_out[data][..., self._internal_output_idx] += x[:, -1, :, :, self._internal_input_idx]
+        return x_out[self._graph_output_meshes[0]]
```

Comment:
Why does this only return the first output mesh?

Reply:
We are returning the first output mesh for compatibility with aifs-mono training. The forward method called in GraphForecaster expects a tensor. I would expect in the future to receive a dict of tensors and return a dict of tensors.